x-pack/plugins/uptime/public/components/overview/kuery_bar/typeahead/typehead.tsx

```diff
@@ -54,210 +39,113 @@ export const Typeahead: React.FC<TypeaheadProps> = ({
   isLoading,
   loadMore,
 }) => {
-  const [state, setState] = useState<TypeaheadState>({
-    isSuggestionsVisible: false,
-    index: null,
-    value: '',
-    inputIsPristine: true,
-    lastSubmitted: '',
-    selected: null,
-  });
+  const [value, setValue] = useState('');
+  const [index, setIndex] = useState<number | null>(null);
+  const [isSuggestionsVisible, setIsSuggestionsVisible] = useState(false);
+
+  const [selected, setSelected] = useState<QuerySuggestion | null>(null);
+  const [inputIsPristine, setInputIsPristine] = useState(true);
+  const [lastSubmitted, setLastSubmitted] = useState('');
+
+  const { kqlSyntax, setKqlSyntax } = useKqlSyntax({ setValue });
 
   const inputRef = useRef<HTMLInputElement>();
 
+  const { setQuery } = useSimpleQuery();
+
   useEffect(() => {
-    if (state.inputIsPristine && initialValue) {
-      setState((prevState) => ({
-        ...prevState,
-        value: initialValue,
-      }));
+    if (inputIsPristine && initialValue) {
+      setValue(initialValue);
     }
 
     // eslint-disable-next-line react-hooks/exhaustive-deps
```

Comment:
Know that this isn't your code, but why are we disabling exhaustive deps here? I know there are real reasons why we may need to do this (I had to in a recent PR), but in general I believe it's an antipattern and advised against by Dan A as it can lead to unintended bugs. If we ever need to do this, we should have a specific comment and also be sure that we've exhausted all our other options.